azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -45,25 +53,52 @@ def __init__(self, msg, inner=None):
         super(InvalidPolicyError, self).__init__(msg, inner)
 
 
+class ExtensionSignaturePolicyError(PolicyError):
+    """
+    Error raised when policy requires signature, but extension is not signed and was not previously validated.
+    This error does not accept a message.
+    """
+    def __init__(self):
+        super(ExtensionSignaturePolicyError, self).__init__()
+
+
+class ExtensionDisallowedError(PolicyError):
+    """
+    Error raised when extension is not present in the policy allowlist.
+    This error does not accept a message.
+    """
+    def __init__(self):
+        super(ExtensionDisallowedError, self).__init__()
+
+
 class _PolicyEngine(object):
     """
     Implements base policy engine API.
     """
     def __init__(self):
         """
-        Initialize policy engine: if policy enforcement is enabled, read and parse policy file.
+        Initialize policy engine with a default policy. This should not raise any errors.
         """
         self._policy_enforcement_enabled = self.__get_policy_enforcement_enabled()
-        self._policy_file_contents = None   # Raw policy file contents will be saved in __read_policy()
+
+        default_policy = {
+            "policyVersion": ustr(_MAX_SUPPORTED_POLICY_VERSION)
+        }
+        self._policy = self._parse_policy(default_policy)
+
+    def update_policy(self, goal_state_history=None):
```

Comment:
do we have scenarios (other than the unit tests) where we do not want to save to the history when updating the policy? if not, let's remove the default value